Makefile

```diff
@@ -287,12 +329,22 @@ ifeq ($(CPU_ONLY), 1)
 	ALL_WARNS := $(ALL_CXX_WARNS)
 	TEST_FILTER := --gtest_filter="-*GPU*"
 	COMMON_FLAGS += -DCPU_ONLY
-endif
+endif	
 
-# Python layer support
-ifeq ($(WITH_PYTHON_LAYER), 1)
-	COMMON_FLAGS += -DWITH_PYTHON_LAYER
-	LIBRARIES += $(PYTHON_LIBRARIES)
+ifeq ($(USE_OPENCL), 1)
+	OBJS := $(PROTO_OBJS) $(CXX_OBJS)
+	TEST_OBJS := $(TEST_CXX_OBJS)
+	TEST_BINS := $(TEST_CXX_BINS)
+	ALL_WARNS := $(ALL_CXX_WARNS)
+	TEST_FILTER := --gtest_filter="*GPU*"
```

Comment:
I don't think this gtest filter is right. Vast majority of tests that run on the GPU don't have the string "GPU" in their name. With this filter, only about 30 tests were being run. To get them all to run I commented this out.